src/Compilers/CSharp/Portable/SymbolDisplay/ObjectDisplay.cs

```diff
@@ -483,33 +483,38 @@ internal static string FormatLiteral(ulong value, ObjectDisplayOptions options)
             return pooledBuilder.ToStringAndFree();
         }
 
-        internal static string FormatLiteral(double value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(double value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
         {
             ValidateOptions(options);
 
-            var result = value.ToString("R", CultureInfo.InvariantCulture);
+            var result = value.ToString("R", GetFormatCulture(cultureInfo));
 
             return options.IncludesOption(ObjectDisplayOptions.IncludeTypeSuffix) ? result + "D" : result;
         }
 
-        internal static string FormatLiteral(float value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(float value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
         {
             ValidateOptions(options);
 
-            var result = value.ToString("R", CultureInfo.InvariantCulture);
+            var result = value.ToString("R", GetFormatCulture(cultureInfo));
 
             return options.IncludesOption(ObjectDisplayOptions.IncludeTypeSuffix) ? result + "F" : result;
         }
 
-        internal static string FormatLiteral(decimal value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(decimal value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
```

Comment:
Maybe have a default value of `CultureInfo.InvariantCulture` instead and save a method invocation and comparasion down the line.